controllers/availabilityController.ts

```diff
@@ -0,0 +1,34 @@
+import { PrismaClient } from ".prisma/client";
+import express, { Request, Response, Router } from "express";
+import AvailabilityRepository from "../models/AvailabilityRepository";
+import { formatDistance, subDays } from "date-fns";
+import formatISO from "date-fns/formatISO";
+export default function AvailabilityController(prismaClient: PrismaClient) {
+  const availibilityRouter = express.Router();
+  const availabilityRepository = new AvailabilityRepository(prismaClient);
+
+  availibilityRouter.get("/availibility/:startTime/:endTime", async (req: Request, res: Response) => {
```

Comment:
Query params vs path params: https://javascript.plainenglish.io/query-strings-url-parameters-d1a35b9a694f

Reply:
Shall I use
// localhost/user?name=max => output: max
app.get(“/user”, (req, res) => {
  res.send(req.query.name)
})
instead of using // localhost/max/23 => output: max 23
app.get(“/user/:name/:age”, (req, res) => {
  res.send(req.params.name + “ “ + req.params.age)
})

Reply:
I was following the example in interviewcontroller. You used the second way I think

Reply:
Generally, for `GET` requests, we use path params only for things like ids and generally single big resource categories maybe. For example, we could have `/user/123` or `/users/active`. 

We generally do not see patterns like `“/user/:name/:age` - because age is not really a nested property of name, and so  these would either be query params `/user?name=abc&age=12`  or we would use `POST` request where { name: abc, age: 12 } would be added to the request body.

Microsoft has some good documentation regarding this: https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md